Realm/Realm.SourceGenerator/InfoClasses.cs

```diff
@@ -0,0 +1,305 @@
+        public virtual NullableAnnotation NullableAnnotation { get; set; } = NullableAnnotation.None;
+
+        public bool HasNullabilityAnnotation => NullableAnnotation == NullableAnnotation.Annotated;
+
+        public string Namespace { get; set; }
+
+        public ITypeSymbol TypeSymbol { get; set; }
+
+        // This includes the eventual nullability annotation
+        public ITypeSymbol CompleteTypeSymbol { get; set; }
+
+        public virtual PropertyTypeInfo InternalType { get; set; }
+
+        public bool IsCollection => CollectionType != null;
+
+        public bool IsListOrSet => IsList || IsSet;
+
+        public bool IsSimpleType => SimpleType != null;
+
+        public bool IsSet => CollectionType == CollectionTypeEnum.Set;
+
+        public bool IsList => CollectionType == CollectionTypeEnum.List;
+
+        public bool IsDictionary => CollectionType == CollectionTypeEnum.Dictionary;
+
+        public bool IsUnsupported => this is UnsupportedTypeInfo;
+
+        public virtual string TypeString => TypeSymbol.ToDisplayString(SymbolDisplayFormat.MinimallyQualifiedFormat);
+
+        public virtual string CompleteTypeString => CompleteTypeSymbol.ToDisplayString(SymbolDisplayFormat.MinimallyQualifiedFormat);
+
+        public static PropertyTypeInfo Unsupported => new UnsupportedTypeInfo();
+
+        public static PropertyTypeInfo List => new ListTypeInfo();
+
+        public static PropertyTypeInfo Set => new SetTypeInfo();
+
+        public static PropertyTypeInfo Dictionary => new DictionaryTypeInfo();
+
+        public static PropertyTypeInfo Int => new SimpleTypeInfo(SimpleTypeEnum.Int);
+
+        public static PropertyTypeInfo Bool => new SimpleTypeInfo(SimpleTypeEnum.Bool);
+
+        public static PropertyTypeInfo String => new SimpleTypeInfo(SimpleTypeEnum.String);
+
+        public static PropertyTypeInfo Data => new SimpleTypeInfo(SimpleTypeEnum.Data);
+
+        public static PropertyTypeInfo Date => new SimpleTypeInfo(SimpleTypeEnum.Date);
+
+        public static PropertyTypeInfo Float => new SimpleTypeInfo(SimpleTypeEnum.Float);
+
+        public static PropertyTypeInfo Double => new SimpleTypeInfo(SimpleTypeEnum.Double);
+
+        public static PropertyTypeInfo Object => new SimpleTypeInfo(SimpleTypeEnum.Object);
+
+        public static PropertyTypeInfo RealmValue => new SimpleTypeInfo(SimpleTypeEnum.RealmValue);
+
+        public static PropertyTypeInfo ObjectId => new SimpleTypeInfo(SimpleTypeEnum.ObjectId);
+
+        public static PropertyTypeInfo Decimal => new SimpleTypeInfo(SimpleTypeEnum.Decimal);
+
+        public static PropertyTypeInfo Guid => new SimpleTypeInfo(SimpleTypeEnum.Guid);
+
+        public static PropertyTypeInfo RealmInteger => new RealmIntegerTypeInfo();
+
+        public static PropertyTypeInfo IQueryable => new IQueryableTypeInfo();
+
+        public bool IsSupportedIndexType()
+        {
+            if (IsRealmInteger)
+            {
+                if (HasNullabilityAnnotation)
```

Comment:
Yes, that is the case. 
We can't use `IsNullable` because of nullability annotation. For example if we have a string:
- Without nullability annotations enabled, `string` will have ` NullableAnnotation == NullableAnnotation.None`
- With nullability annotations enabled, `string` will have ` NullableAnnotation == NullableAnnotation.NotAnnotated`